redfish-core/lib/assembly.hpp

```diff
@@ -252,29 +252,37 @@ void getAssemblyPresence(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
                          const auto& serviceName, const auto& assembly,
                          const auto& assemblyIndex)
 {
-    nlohmann::json& assemblyArray = asyncResp->res.jsonValue["Assemblies"];
-    nlohmann::json& assemblyData = assemblyArray.at(assemblyIndex);
-
-    assemblyData["Status"]["State"] = "Enabled";
-
     sdbusplus::asio::getProperty<bool>(
         *crow::connections::systemBus, serviceName, assembly,
         "xyz.openbmc_project.Inventory.Item", "Present",
-        [asyncResp, assemblyIndex](const boost::system::error_code& ec,
-                                   const bool value) {
+        [asyncResp, assemblyIndex,
+         assembly](const boost::system::error_code& ec, const bool value) {
         if (ec)
         {
             BMCWEB_LOG_ERROR("DBUS response error: {}", ec.value());
             messages::internalError(asyncResp->res);
             return;
         }
 
+        nlohmann::json& array = asyncResp->res.jsonValue["Assemblies"];
+        nlohmann::json& data = array.at(assemblyIndex);
+        data["Status"]["State"] = "Enabled";
         if (!value)
         {
-            nlohmann::json& array = asyncResp->res.jsonValue["Assemblies"];
-            nlohmann::json& data = array.at(assemblyIndex);
             data["Status"]["State"] = "Absent";
         }
+
+        std::string fru = sdbusplus::message::object_path(assembly).filename();
+        // Special handling for LCD and base panel CM.
+        if (fru == "panel0" || fru == "panel1")
+        {
+            data["Oem"]["OpenBMC"]["@odata.type"] =
+                "#OemAssembly.v1_0_0.Assembly";
+
+            // if panel is not present, implies it is already removed or can be
+            // placed.
+            data["Oem"]["OpenBMC"]["ReadyToRemove"] = value;
```

Comment:
Does this `ReadyToRemove` have a default value?
If `getProperty()` has failed, `ReadyToRemove` is not filled up.  Should it set to 'true' if so?

https://github.com/ibm-openbmc/bmcweb/blob/657cf7caa63942d83c45945a2dc158392bc667f8/redfish-core/lib/assembly.hpp#L213C1-L249C2

```
void getAssemblyPresence(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
                         const auto& serviceName, const auto& assembly,
                         const auto& assemblyIndex)
{
    nlohmann::json& assemblyArray = asyncResp->res.jsonValue["Assemblies"];
    nlohmann::json& assemblyData = assemblyArray.at(assemblyIndex);

    assemblyData["Status"]["State"] = "Enabled";

   sdbusplus::asio::getProperty<bool>(
        *crow::connections::systemBus, serviceName, assembly,
        "xyz.openbmc_project.Inventory.Item", "Present",
        [asyncResp, assemblyIndex,
         assembly](const boost::system::error_code& ec, const bool value) {
...
        if (ec)
        {
            //===> Would this be needed?
            if (ec.value() == boost::system::errc::io_error)
            {
                    nlohmann::json& array = asyncResp->res.jsonValue["Assemblies"];
                    nlohmann::json& data = array.at(assemblyIndex);
                    data["Oem"]["OpenBMC"]["@odata.type"] = "#OemAssembly.v1_0_0.Assembly";
                    data["Oem"]["OpenBMC"]["ReadyToRemove"] = true;
                    return;
            }
            BMCWEB_LOG_ERROR("DBUS response error: {}", ec.value());
            messages::internalError(asyncResp->res);
            return;
        }
```

Reply:
please check new code